nomad/event_sink_manager.go

```diff
@@ -259,17 +262,15 @@ func (m *SinkManager) refreshSinks() error {
 	return nil
 }
 
-// NewSinkWs returns the current newSinkWs used to listen for changes to the
-// event sink table in the state store
-func (m *SinkManager) NewSinkWs() memdb.WatchSet {
-	m.mu.Lock()
-	defer m.mu.Unlock()
+// Ws returns the current newSinkWs used to listen for changes to the
+// event sink table in the state store. Ws() should only be called from Run
+func (m *SinkManager) Ws() memdb.WatchSet {
```

Comment:
If this should only be called from `SinkManager.Run` (for synchronization), maybe it should be non-exported or just inlined.

Reply:
will update to be non-exported